streams/src/main/java/org/apache/kafka/streams/kstream/internals/metrics/Sensors.java

```diff
@@ -38,13 +45,89 @@ public static Sensor lateRecordDropSensor(final InternalProcessorContext context
         );
         StreamsMetricsImpl.addInvocationRateAndCount(
             sensor,
-            "stream-processor-node-metrics",
+            PROCESSOR_NODE_METRICS_GROUP,
             metrics.tagMap("task-id", context.taskId().toString(), "processor-node-id", context.currentNode().name()),
             "late-record-drop"
         );
         return sensor;
     }
 
+    public static Sensor suppressionSensor(final InternalProcessorContext context) {
+        final StreamsMetricsImpl metrics = context.metrics();
+        final String operation = "intermediate-result-suppression";
+
+        final Sensor sensor = metrics.nodeLevelSensor(
+            context.taskId().toString(),
+            context.currentNode().name(),
+            operation,
+            Sensor.RecordingLevel.DEBUG
+        );
+
+        final Map<String, String> tags = metrics.tagMap(
+            "task-id", context.taskId().toString(),
+            "processor-node-id", context.currentNode().name()
+        );
+
+        sensor.add(
+            new MetricName(
+                operation + "-rate",
+                PROCESSOR_NODE_METRICS_GROUP,
+                "The average number of occurrence of " + operation + " operation per second.",
```

Comment:
nit: Should the string say `The rate of...`?

Reply:
sure!

Reply:
Actually, strange as the sentence is, this is what the other rate metrics say, so I think we should just keep it for consistency.